heidrun/nara_json.rb

```diff
@@ -0,0 +1,575 @@
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/nara'
+    label 'National Archives and Records Administration'
+  end
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('description', 'item | itemAv | fileUnit',
+                               'physicalOccurrenceArray',
+                               'itemPhysicalOccurrence |' \
+                                 'itemAvPhysicalOccurrence |' \
+                                 'fileUnitPhysicalOccurrence',
+                               'referenceUnitArray', 'referenceUnit',
+                               'name')
+  end
+
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('naId').first_value.map { |id| nara_catalog_uri(id) }
+  end
+
+  object :class => DPLA::MAP::WebResource do
+    uri record.field('objects', 'object', 'file').first_value
+              .map { |o| make_obj_uri(o) }
+    dcformat record.field('objects', 'object', 'file').first_value
+                   .map { |o| make_obj_dcformat(o) }
+  end
+
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('objects', 'object', 'thumbnail').first_value
+              .map { |o| make_preview_uri(o) }
+    dcformat record.field('objects', 'object', 'thumbnail').first_value
+                   .map { |o| make_preview_dcformat(o) }
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+
+    # <parentRecordGroup> OR <parentCollection>
+    #   <naId>[VALUE]</naId>
+    #   <title>[VALUE]</title>
+    #   <recordGroupNumber>[VALUE]</recordGroupNumber>
+    # </parentRecordGroup> OR </parentCollection>
+    collection :class => DPLA::MAP::Collection do
+      title record.field('description', 'item | itemAv | fileUnit',
+                         'parentSeries',
+                         'parentRecordGroup | parentCollection', 'title')
+    end
+
+    contributor :class => DPLA::MAP::Agent do
+      providedLabel record.field('description', 'item | itemAv | fileUnit',
+                                 'organizationalContributorArray |' \
+                                   'personalContributorArray')
+                          .map { |el| make_contributor(el) }
+    end
+
+    # *Use <contributorType>Most Recent</contributorType> if multiple <contributor-display> values.
+    #
+    # FIXME:  the instruction on the line above doesn't make sense with the
+    #         structure of this element in the original data. Verify?
+    #
+    # <creatingOrganizationArray> OR <creatingIndividualArray>
+    #   <creatingOrganization> OR <creatingIndividual>
+    #     <creator>
+    #       <termName>[VALUE]</termName>
+    #     </creator>
+    #   </creatingOrganization> OR </creatingIndividual>
+    # </creatingOrganizationArray> OR </creatingIndividualArray>
+    creator :class => DPLA::MAP::Agent do
```

Comment:
I'll create a Gist.

I have also reverted Krikri to v0.5.7, which was before I made the mapping change to add the "or" operator, and I get the same result as pasted above, with repeated arrays.  (I modified the Gemfile and reinstalled the bundle.  Then I commented out everything in the mapping except for the `creator` section.  I modified the parameters to `field` to get rid of the `|` conditions, and ran the mapping manually on the same record that I'd used above.)